compaction/task_manager_module.cc

```diff
@@ -423,6 +423,10 @@ future<> table_major_keyspace_compaction_task_impl::run() {
 
 future<> cleanup_keyspace_compaction_task_impl::run() {
     co_await _db.invoke_on_all([&] (replica::database& db) -> future<> {
+        if (_do_flush_tables) {
+            // this task could be a part of a batch, which flushes tables by itself
+            co_await db.flush_all_tables();
+        }
         auto& module = db.get_compaction_manager().get_task_manager_module();
         auto task = co_await module.make_and_start_task<shard_cleanup_keyspace_compaction_task_impl>({_status.id, _status.shard}, _status.keyspace, _status.id, db, _table_infos);
```

Comment:
If all tables are flushed we should pass a flag downstream to not flush tables in `perform_cleanup_compaction`, similar to what we do in major compaction.
Can be done in a separate patch, but I'd like it to be included in this series.